pkgs/by-name/ne/netbox_4_0/package.nix

```diff
@@ -0,0 +1,118 @@
+{
+  lib,
+  fetchFromGitHub,
+  python3,
+  plugins ? _ps: [ ],
+  nixosTests,
+}:
+let
+  py = python3.override { packageOverrides = _final: prev: { django = prev.django_5; }; };
+
+  extraBuildInputs = plugins py.pkgs;
+in
+py.pkgs.buildPythonApplication rec {
+  pname = "netbox";
+  version = "4.0.11";
+
+  format = "other";
+
+  src = fetchFromGitHub {
+    owner = "netbox-community";
+    repo = pname;
+    rev = "refs/tags/v${version}";
+    hash = "sha256-0yEz7v5RL1+cqbGDyuyEsywFonJQfPdVIQdL0qLyc04=";
+  };
+
+  patches = [ ./custom-static-root.patch ];
+
+  propagatedBuildInputs =
+    (with py.pkgs; [
+      django
+      django-cors-headers
+      django-debug-toolbar
+      django-filter
+      django-graphiql-debug-toolbar
+      django-htmx
+      django-mptt
+      django-pglocks
+      django-prometheus
+      django-redis
+      django-rq
+      django-tables2
+      django-taggit
+      django-timezone-field
+      djangorestframework
+      drf-spectacular
+      drf-spectacular-sidecar
+      feedparser
+      jinja2
+      markdown
+      netaddr
+      nh3
+      pillow
+      psycopg2
+      pyyaml
+      requests
+      social-auth-core
+      social-auth-app-django
+      strawberry-graphql
+      strawberry-django
+      svgwrite
+      tablib
+
+      # Optional dependencies, kept here for backward compatibility
+
+      # for the S3 data source backend
+      boto3
+      # for Git data source backend
+      dulwich
+      # for error reporting
+      sentry-sdk
+    ])
```

Comment:
`python-jose` is also needed for OIDC auth.

Reply:
Good catch! I've added it through `social-auth-core.passthru.optional-dependencies.openidconnect`